src/abstract_register.jl

```diff
@@ -163,44 +176,68 @@ end
 
 ## Measurement
 
+export ComputationalBasis
+struct ComputationalBasis end
+
 """
-    measure(register[, locs]; nshots=1) -> Vector{Int}
+    measure([operator], register[, locs]; nshots=1) -> Vector{Int}
```

Comment:
If you want to add an optional keyword as first argument, you need to write two methods, or the overloading won't be correct.

Reply:
And if you define this interface, which method is supposed to overload? Are we going to implement `measure(register)` and `measure(op, register)` separately or just let op be compute basis by default?

Reply:
```
measure(::ComputationalBasis, reg, ::AllLocs)
```

Is the one should be implemented, the reset can fall back.

I changed the title to WIP.